extras/heavy_em_n3lo/yad_grids.py

```diff
@@ -11,15 +11,35 @@
 
 nf = int(sys.argv[1])
 n_threads = int(sys.argv[2])
-channel = sys.argv[3]
-order = int(sys.argv[4])
-try:
-    variation = int(sys.argv[5])
-except IndexError:
-    variation = 0
+kind = sys.argv[3]
+if kind not in ['2', 'L']:
+    raise ValueError("Set kind to '2' or 'L'")
+
+channel = sys.argv[4]
+if channel not in ['q', 'g']:
+    raise ValueError("Set channel to 'g' or 'q'")
+
+order = int(sys.argv[5])
+
+hs_version = sys.argv[6]
+if hs_version not in ["klmv", "abmp", "abbdvss"]:
+    raise ValueError("Set hs_version to 'abmp' or 'abbdvss'")
+# klmv = Kawamura, Lo Presti, Moch, Vogt: approximation from [arXiv:1205.5727]
+# abmp = Alekhin, Blumlein, Moch, Placakyte: approximation from [arXiv:1701.05838]
+# abbdvss = Ablinger, Behring, Blumlein, De Freitas, von Manteuffel, Schneider, Schonwald: approximation from [arXiv:2403.00513]
```

Comment:
The approximation is actually mine, based on their result, but not exactly them... 
(I mention this before someone will complain, not because I care of the authorship)

Reply:
ok sure. tomorrow I will fix both in yadism and adani